packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -222,10 +225,66 @@ class UpgradeCommandRunner {
     }
   }
 
+  /// Checks if the Flutter git repository is tracking a "standard remote".
+  ///
+  /// Using `flutter upgrade` is not supported from a non-standard remote. A git
+  /// remote should have the same url as [_flutterGit] to be considered as a
+  /// "standard" remote.
+  ///
+  /// Exits tool if the tracking remote is not standard.
+  void verifyStandardRemote(FlutterVersion localVersion) {
+    // Strip `.git` suffix from repository url and _flutterGit
+    final String trackingUrl = _stripDotGit(localVersion.repositoryUrl);
```

Comment:
I realize that we also should support upstreams configured of the form: `git@github.com:flutter/flutter.git`.

Reply:
This shouldn't affect anything? HTTPS or SSH urls should behave the same whether they have `.git` or not, which is what `_stripDotGit` is concerned with.

Reply:
When I run this with `git@github.com:flutter/flutter.git` configured, the tool exits saying that I have a non-standard remote. This should be considered a standard remote.

Reply:
So what I understand is for comparing two urls regardless of whether they are HTTPS or SSH, we need to check two things:
 - The domain should be the same.
 - The path should be the same.
 
After using `_stripDotGit`, we could pass both strings into this function:

```dart
bool _checkSameRemote(String url1, String url2) {
  RegExp pattern = RegExp(r'(.*)(\/\/|@)(www\.)?([A-Za-z0-9.-]+)(:|\/)(.*)$');
  final RegExpMatch match1 = pattern.firstMatch(url1);
  final RegExpMatch match2 = pattern.firstMatch(url2);
  String url1Domain = match1?.group(4);
  String url2Domain = match2?.group(4);
  String url1Path = match1?.group(6);
  String url2Path = match2?.group(6);
  if (url1Domain == url2Domain && url1Path == url2Path) {
    return true;
  } else {
    return false;
  }
}
```
[I've played around with this implementation in a dartpad](https://dartpad.dev/1256fcc3049d6da068ee3e8433df5b02). Seems to be doing its job as far as I can tell.

Reply:
Additional discussion at https://discord.com/channels/608014603317936148/608022056616853515/838804935691993178